app/src/main/java/com/simplecity/amp_library/ShuttleApplication.java

```diff
@@ -197,6 +209,14 @@ public void onCreate() {
                 .onErrorComplete()
                 .subscribeOn(Schedulers.io())
                 .subscribe();
+
+        currentVolume = Objects.requireNonNull(HeadphoneUtils.getVolumeData(mApp)).getCurrentVolumeLevel();
+        MediaRouter mediaRouter = (MediaRouter) getApplicationContext().getSystemService(MEDIA_ROUTER_SERVICE);
```

Comment:
https://developer.android.com/guide/topics/media/mediarouter says:

> Content sent over a media route passes through the route's associated MediaRouteProvider (except in a few special cases, such as a Bluetooth output device).

So would we not be able to get volume events from a Bluetooth device?

Could you ask Dr. Meng if he can get you access to a Bluetooth speaker or headset so you can test?

Reply:
I have access to a bluetooth device, I could use that to test this out.